catbuffer/parser/tests/generators/test_util.py

```diff
@@ -247,5 +251,117 @@ def test_extends_size_reference_field_with_bound_size(self):
 		self._assert_field_extensions(models[1].fields[0].extensions, 'Bar', 'Bar beta (custom)', size_field_names=['beta_size'])
 		self._assert_field_extensions(models[1].fields[1].extensions, 'beta_size', 'beta_size beta_size (pod)', bound_field_name='beta')
 
+	@staticmethod
+	def create_aligned_struct(name):
+		struct = Struct([None, name])
+		struct.attributes = [
+			Attribute(['is_aligned'])
+		]
+		return struct
+
+	@staticmethod
+	def create_struct_with_factory(name, factory_name, fields=None):
+		struct = Struct([None, name])
+		if fields:
+			struct.fields = fields
+		struct.factory_type = factory_name
+		return struct
+
+	def _assert_unaligned(self, models, models_unaligned):
+		for model in models:
+			self.assertEqual(model in models_unaligned, model.requires_unaligned)
+
+	def test_extends_marks_model_when_aligned_used_in_non_aligned(self):
+		# Arrange:
+		models = [
+			self.create_aligned_struct('Foo'),
+			Struct([None, 'Bar']),
+			Struct([
+				None,
+				'FooContainer',
+				StructField(['foo', Array(['Foo', 5])]),
+				StructField(['bar', Array(['Bar', 5])]),
+			])
+		]
+
+		# Act:
+		extend_models(models, self._printer_factory)
+
+		# Assert:
+		self._assert_unaligned(models, [models[0]])
+
+	def test_extends_marks_derived_models_when_aligned_is_a_factory(self):
+		# Arrange:
+		models = [
+			self.create_struct_with_factory('Foox', 'FooBase'),
+			self.create_struct_with_factory('Fooy', 'FooBase'),
+			self.create_aligned_struct('FooBase'),
+			Struct([
+				None,
+				'FooContainer',
+				StructField(['foos', Array(['FooBase', 5])]),
+			])
+		]
+
+		# Act:
+		extend_models(models, self._printer_factory)
+
+		# Assert:
+		self._assert_unaligned(models, models[0:3])
+
+	def test_extends_marks_struct_models_used_as_fields_in_derived_models(self):
+		# Arrange:
+		models = [
+			self.create_aligned_struct('Alpha'),
+			self.create_aligned_struct('Beta'),
+			self.create_aligned_struct('Gamma'),
+			self.create_struct_with_factory('Foo', 'FooBase', [
+				StructField(['alpha', 'Alpha']),
+			]),
+			self.create_struct_with_factory('Bar', None, [
+				StructField(['gamma', 'Gamma']),
+			]),
+			self.create_aligned_struct('FooBase'),
+			Struct([
+				None,
+				'FooContainer',
+				StructField(['foos', Array(['FooBase', 5])]),
+			])
+		]
+
+		# Act:
+		extend_models(models, self._printer_factory)
+
+		# Assert: following structs are supposed to have unaligned attribute
+		#  - FooBase
+		#  - Foo - cause derived from FooBase and FooBase used inside array
+		#  - Alpha - cause used inside Foo
+		self._assert_unaligned(models, [models[0], models[3], models[5]])
+
+	def test_extends_marking_is_recursive(self):
+		# Arrange:
+		models = [
+			Struct([None, 'Alpha']),
```

Comment:
any reason this one is not with is_allgned attribute?

Reply:
shouldn't really matter, but yeah, changed